doc/conf.py

```diff
@@ -330,9 +331,6 @@ def setup(app):
     'doc_module': ('sphinx_gallery', 'numpy'),
     'reference_url': {
         'sphinx_gallery': None,
```

Comment:
The `sphinx_gallery` looks a bit akward on its own, is it really needed and is there a way we can get rid of `reference_url` all together and only use intersphinx?

Reply:
Indeed, this becomes redundant with intersphinx available. I don't even think we need it. The `None` is there to tell sphinx to find the docs database locally. But we can have that done at execution time.

Reply:
Ahh I see what you're saying, I can make a quick PR for that.

Reply:
... but you still need to at least get `this_module` name from somewhere, otherwise I'm not sure how else you'd set the `doc_resolvers[this_module] = ...`. Without that, we don't know if we should look locally or to use `intersphinx`.

I don't see the module name in the `config` anywhere else. Storing it as a `dict` instead of a single string entry now is a bit silly, as is the name `reference_url` instead of something like `root_module` or something, but I don't mind the current version too much.

Reply:
I don't consider this very urgent. There is indeed some redundancy. And having now a dictionary for a single entry is no too elegant.
My idea is that one usually documents one local module and that module is in the "doc_module" value. Since there isn't now any need of specifying all other "reference_url" for external modules, as `intersphinx` take care of it. We could iterate the values of "doc_module" to see which modules we should try to resolve locally. 
I don't know what will happen in the case of sphinx-gallery that documents `numpy` too track the use of it in the code examples and generate the mini-galleries. I know is not related, but using the values of `doc_module` intersphinx will try to search for numpy locally and fail, and then totally independent, the value of interphinx sources in `conf.py` will take over?